publisher/elbv2.go

```diff
@@ -0,0 +1,122 @@
+package publisher
+
+import (
+  "bufio"
+  "fmt"
+  "math/rand"
+  "os"
+  "runtime"
+  "strings"
+
+  "github.com/Sirupsen/logrus"
+  dynsampler "github.com/honeycombio/dynsampler-go"
+  "github.com/honeycombio/honeyaws/state"
+  "github.com/honeycombio/honeytail/event"
+  "github.com/honeycombio/honeytail/parsers/nginx"
+)
+
+type ELBv2EventParser struct {
+  sampler dynsampler.Sampler
+}
+
+func NewELBv2EventParser(sampleRate int) *ELBv2EventParser {
+  ep := &ELBv2EventParser{
+    sampler: &dynsampler.AvgSampleRate{
+      ClearFrequencySec: 300,
+      GoalSampleRate:    sampleRate,
+    },
+  }
+
+  if err := ep.sampler.Start(); err != nil {
+    logrus.WithField("err", err).Fatal("Couldn't start dynamic sampler")
+  }
+
+  return ep
+}
+
+func (ep *ELBv2EventParser) ParseEvents(obj state.DownloadedObject, out chan<- event.Event) error {
+  np := &nginx.Parser{}
+  err := np.Init(&nginx.Options{
+    ConfigFile:      formatFileNameV2,
+    TimeFieldName:   "timestamp",
+    TimeFieldFormat: "2006-01-02T15:04:05.9999Z",
+    LogFormatName:   AWSElasticLoadBalancerFormatV2,
+    NumParsers:      runtime.NumCPU(),
+  })
+  if err != nil {
+    logrus.Fatal("Can't initialize the nginx parser")
+  }
+
+  linesCh := make(chan string)
+
+  go np.ProcessLines(linesCh, out, nil)
+  f, err := os.Open(obj.Filename)
```

Comment:
Here is where you want to move the un-gzip code to. Just create a `gzip.NewReader` for the downloaded gzipped object.